config/config.go

```diff
@@ -139,24 +142,28 @@ func UpdateGlobal(f func(conf *Config)) func() {
 	return restore
 }
 
-const (
-	globalTxnScope = "global"
-)
-
-// GetTxnScopeFromConfig extracts @@txn_scope value from config
+// GetTxnScopeFromConfig returns whether it's global scope and extracts @@txn_scope value from config.
 func GetTxnScopeFromConfig() (bool, string) {
+	var (
+		enableLocalTxn bool
+		txnScope       string
+	)
+	if kvcfg := GetGlobalConfig(); kvcfg != nil {
+		enableLocalTxn = kvcfg.EnableLocalTxn
+		txnScope = kvcfg.TxnScope
+	}
+	if val, err := util.EvalFailpoint("injectEnableLocalTxn"); err == nil {
+		enableLocalTxn = val.(bool)
+	}
 	if val, err := util.EvalFailpoint("injectTxnScope"); err == nil {
-		v := val.(string)
-		if len(v) > 0 {
-			return false, v
-		}
-		return true, globalTxnScope
+		txnScope = val.(string)
 	}
 
-	if kvcfg := GetGlobalConfig(); kvcfg != nil && len(kvcfg.TxnScope) > 0 {
-		return false, kvcfg.TxnScope
+	// If EnableLocalTxn is false, we will always return it's global.
+	if enableLocalTxn && len(txnScope) > 0 {
+		return false, txnScope
 	}
-	return true, globalTxnScope
+	return true, oracle.GlobalTxnScope
```

Comment:
Does this affect the stale read? https://github.com/pingcap/tidb/pull/25525/files#diff-ed09db7ac6827940f6b3ac9d72565c4fea1136d8e6ff76ad7aa7ffefc73aca26R1535

Reply:
cc @Yisaer